src/diagnostics/land_compute_methods.jl

```diff
@@ -0,0 +1,145 @@
+"""
+    @land_compute name model compute
+
+
+Macro generating a function to compute a land diagnostic,
+needed in the ClimaDiagnostics framework.
+
+For example,
+@land_compute "soil_net_radiation" SoilCanopyModel p.soil.R_n
+
+generates the function
+
+function compute_soil_net_radiation!(out, Y, p, t, land_model::SoilCanopyModel)
+    if isnothing(out)
+        return copy(p.soil.R_n)
+    else
+        out .= p.soil.R_n
+    end
+end
+"""
+macro land_compute(name, model, compute)
+    functions = Expr[]
+    function_name = Symbol("compute_", name, "!")
+    function_body = esc(quote
+        function $function_name(out, Y, p, t, land_model::$model)
+            if isnothing(out)
+                return copy($compute)
+            else
+                out .= $compute
+            end
+        end
+    end)
+    push!(functions, function_body)
+    return quote
+        $(functions...)
+    end
+end
+
+### BucketModel ###
+
+# variables stored in p (diagnostics variables stored in the cache)
+@land_compute "aerodynamic_resistance" BucketModel p.bucket.turbulent_fluxes.r_ae
+@land_compute "albedo" BucketModel p.bucket.α_sfc
```

Comment:
For this we want to be specific about the band. Is it shortwave (400-2500nm), PAR or NIR?

Reply:
@kmdeck

Reply:
For the bucket, it is a bulk shortwave albedo